cpp/src/arrow/compute/exec/hash_join_node.cc

```diff
@@ -494,7 +494,13 @@ class HashJoinNode : public ExecNode {
     size_t thread_index = thread_indexer_();
     int side = (input == inputs_[0]) ? 0 : 1;
     {
-      Status status = impl_->InputReceived(thread_index, side, std::move(batch));
+      auto batch = task();
+      if (!batch.ok()) {
+        StopProducing();
+        ErrorIfNotOk(batch.status());
+        return;
+      }
+      Status status = impl_->InputReceived(thread_index, side, batch.MoveValueUnsafe());
```

Comment:
This would change to submit task.

Reply:
That's right, that change will happen when in the PRs related to the scheduler x2 :)